### ISSUE TITLE
<hr> separator isn't parsed anymore in new posts

### ISSUE THREAD

**Issue by ThunderAxe31** [TASVideos/tasvideos#831]
You can see some horizontal line separators in this first post of this old thread: https://tasvideos.org/Forum/Topics/18754
If you hit the Edit button for that post, you can clearly see that those lines were in fact generated with `<hr>` text.

But in new posts, the very same `<hr>` do nothing:
![image](https://user-images.githubusercontent.com/36802254/148355609-81c467ba-51c5-4b72-9001-926ffd32ee8e.png)

This may lead to awkward situations in which by editing an old forum post, you kill the horizontal lines and destroy its original layout.

**Comment by nattthebear** [comment]
This is intended.  The old forums had an allowHTML flag that allowed a restricted subset of HTML to be parsed.  The new ones [support it](https://github.com/TASVideos/tasvideos/blob/main/TASVideos.ForumEngine/BbParser.cs#L382) for backcompat, but not in new posts so as to limit our maintenance burden going forward.

**Comment by ThunderAxe31** [comment]
All righty then, feel free to add the label "as intended" to this issue and close it.

**Comment by vadosnaprimer** [comment]
What about having a `[hr]` forum tag then? There really are really cases when it's needed.

**Comment by nattthebear** [comment]
Leaving open to get more feedback -- @adelikat and I went over this a year or two back, but haven't revisited it since then.

**Comment by TiKevin83** [comment]
IMO having the backcompat support but then not using it going forward is a recipe for complications like this, it may be less of a tech debt burden to just continue supporting the HTML backcompat subset

**Comment by nattthebear** [comment]
> complications like this

To be fair, this isn't much of a complication.  Old posts still work fine, new posts work fine, and editing old posts is generally limited to recent posts.  As time goes on, this issue will matter less and less.

**Comment by ThunderAxe31** [comment]
Well, I personally don't see why `<hr>` should be blocked for new posts. What harm could it ever cause?

> What about having a `[hr]` forum tag then? There really are cases when it's needed.

Now that is a complication. If we want to allow it, then just allow it by writing `<hr>` in a post, and keep ignoring any other kind of html code.

**Comment by Invariel** [comment]
If [hr] is included as a parsed Forum code, it causes fewer problems than allowing <hr>.  Theoretically, we could even support something the width parameter so that you don't have <hr> tags that span the entire textbox.

tl;dr: Adding in an edge case for <hr> offers a hole for exploitation that adding support for [hr] does not.

**Comment by nattthebear** [comment]
The parsers (both forum and wiki) are homegrown handmade creations by me.  I am... kind of okish at writing these, but not great.  I want to keep the syntax space down, and the number of things we need to support for updates and changes down.  All complexity is harm, and the most important long term goal of this site is to keep it maintainable by non-wizards.

**Comment by adelikat** [comment]
@nattthebear : I support having a [hr] tag

**Comment by nattthebear** [comment]
Are we OK with `[hr][/hr]`?  It feels kind of weird because there's no content to put inside it, but at the same time the parser currently has no notion of a void bbcode tag.

**Comment by vadosnaprimer** [comment]
Does it have to require a closing tag? `<hr>` takes the entire line so it feels right for `[hr]` to also just take the entire line.
